### PR TITLE
Docs update

### LINKED ISSUE #233: Ensure readme for examples are up to date

### PR DESCRIPTION
**Pull request by TimEvens**
1. Fixed issue with docs where docs did not include the inherited members
2. Renamed docs to not include moq
3. Grouped callbacks in their own section in the docs so that it's clear which methods are callbacks 
4. Added api guide documentation, starting with high level overview and client
5. Added examples, specifically qserver and qclient

Resolves #233

### CODE REVIEW

File: include/quicr/client.h [hunk 1]
```diff
@@ -54,6 +54,12 @@ namespace quicr {
          */
         Status Disconnect();
 
+        // --BEGIN CALLBACKS ----------------------------------------------------------------------------------
+        /** @name Client Calbacks
+         *      client transport specific callbacks
+         */
+        ///@{
```
Comment (GhostofCookie) [inline include/quicr/client.h:61]: What is this notation?

File: include/quicr/publish_track_handler.h [hunk 2]
```diff
@@ -107,6 +107,9 @@ namespace quicr {
         // --------------------------------------------------------------------------
         // Public Virtual API callback event methods
         // --------------------------------------------------------------------------
+        /** @name Calbacks
```
Comment (GhostofCookie) [inline include/quicr/publish_track_handler.h:110]:
```suggestion
        /** @name Callbacks
```

File: include/quicr/server.h [hunk 3]
```diff
@@ -74,6 +74,40 @@ namespace quicr {
...
+         *
+         * @param connection_handle        source connection ID
+         * @param subscribe_id             subscribe ID
+         * @param subscribe_response       response to for the subscribe
+         */
+        virtual void ResolveSubscribe(ConnectionHandle connection_handle,
+                                      uint64_t subscribe_id,
+                                      const SubscribeResponse& subscribe_response);
+
+        // --BEGIN ALLBACKS ----------------------------------------------------------------------------------
```
Comment (GhostofCookie) [inline include/quicr/server.h:105]:
```suggestion
        // --BEGIN CALLBACKS ----------------------------------------------------------------------------------
```

File: include/quicr/subscribe_track_handler.h [hunk 4]
```diff
@@ -77,6 +77,9 @@ namespace quicr {
         // --------------------------------------------------------------------------
         // Public Virtual API callback event methods
         // --------------------------------------------------------------------------
+        /** @name Calbacks
```
Comment (GhostofCookie) [inline include/quicr/subscribe_track_handler.h:80]:
```suggestion
        /** @name Callbacks
```

File: src/server.cpp [hunk 5]
```diff
@@ -39,6 +39,57 @@ namespace quicr {
...
+
+        track_handler->publish_data_ctx_id_ =
+          quic_transport_->CreateDataContext(conn_id,
+                                             track_handler->default_track_mode_ == TrackMode::kDatagram ? false : true,
+                                             track_handler->default_priority_,
+                                             false);
+
+        // Setup the function for the track handler to use to send objects with thread safety
+        track_handler->publish_object_func_ =
+          [&, track_handler = track_handler, subscribe_id = track_handler->GetSubscribeId()](
```
Comment (GhostofCookie) [inline src/server.cpp:76]: I think you can simply write `track_handler` to copy it, but this notation doesn't hurt.

File: include/quicr/client.h (hunk 1 above)
Comment (TimEvens) [reply include/quicr/client.h:61]: specific to doxygen... https://www.doxygen.nl/manual/grouping.html

File: include/quicr/publish_track_handler.h (hunk 2 above)
Comment (TimEvens) [reply include/quicr/publish_track_handler.h:110]: updated.

File: include/quicr/server.h (hunk 3 above)
Comment (TimEvens) [reply include/quicr/server.h:105]: updated

File: include/quicr/subscribe_track_handler.h (hunk 4 above)
Comment (TimEvens) [reply include/quicr/subscribe_track_handler.h:80]: Updated

File: src/server.cpp (hunk 5 above)
Comment (TimEvens) [reply src/server.cpp:76]: Yeah, we can look at changing that later.

**Review by GhostofCookie — APPROVED** [review]

**Pull request merged**